cms/core/tests/test_video_embed_block.py

```diff
@@ -0,0 +1,254 @@
+from django.test import TestCase
+from wagtail.blocks import StructBlockValidationError
+from wagtail_factories import ImageFactory
+
+from cms.core.blocks import (
+    VideoEmbedBlock,
+)
+
+
+class VideoEmbedBlockTestCase(TestCase):
+    """Test for video embed block."""
+
+    def test_videoembedblock_clean__invalid_domain(self):
+        """Check the VideoEmbedBlock validates the supplied URL."""
+        block = VideoEmbedBlock()
+        image = ImageFactory.create()
+
+        with self.assertRaises(StructBlockValidationError) as info:
+            value = block.to_python(
+                {
+                    "link_url": "https://ons.gov.uk/908205163",
+                    "image": image.id,
+                    "title": "The video",
+                    "link_text": "Watch the video",
+                }
+            )
+            block.clean(value)
+
+        self.assertEqual(
+            info.exception.block_errors["link_url"].message,
+            "The link URL must use a valid vimeo or youtube video URL",
+        )
+
+    def test_videoembedblock_clean__invalid_youtube_link_1(self):
```

Comment:
non-blocking: in the future you could use `subTest` for these: https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests